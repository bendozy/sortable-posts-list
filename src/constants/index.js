export const MOVE_POST = 'MOVE_POST';
export const UP = 'UP';
export const DOWN = 'DOWN';
export const FETCH_POSTS = 'FETCH_POSTS';
export const FETCH_POSTS_SUCCESS = 'FETCH_POSTS_SUCCESS';
export const FETCH_POSTS_ERROR = 'FETCH_POSTS_ERROR';
