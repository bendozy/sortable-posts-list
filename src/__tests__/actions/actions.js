import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import timeTravel, { removeLastAction } from '../../actions/actions';
import { REMOVE_LAST_ACTION } from '../../constants';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.useFakeTimers();

describe('Action creators tests', () => {
  it('should create an action to dispatch last item in the actions reducer', () => {
    const lastAction = {
      title: 'hello',
      from: 1,
    };
    const expectedAction = {
      type: REMOVE_LAST_ACTION,
      lastAction,
    };

    expect(removeLastAction(lastAction)).toEqual(expectedAction);
    expect(removeLastAction(lastAction)).toMatchSnapshot();
  });

  it('dispatches only one action if there is only an action Item in the actions reducer', () => {
    const lastAction = {
      title: 'hello',
      from: 1,
    };
    const expectedAction = {
      type: REMOVE_LAST_ACTION,
      lastAction,
    };

    const store = mockStore({
      actions: {
        actions: [lastAction],
      },
    });

    store.dispatch(timeTravel(0));

    expect(store.getActions()).toContainEqual(expectedAction);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches only all actions in the actions reducer if index 0 is selected', () => {
    const actions = Array.from({ length: 5 }, (v, i) => ({ index: i }));
    const type = REMOVE_LAST_ACTION;

    const store = mockStore({
      actions: { actions },
    });

    store.dispatch(timeTravel(0));
    jest.runAllTimers();

    expect(store.getActions()).toHaveLength(actions.length);
    expect(store.getActions(store.getActions().length - 1))
      .toContainEqual({ type, lastAction: actions[0] });
    expect(store.getActions(0))
      .toContainEqual({ type, lastAction: actions[actions.length - 1] });
    expect(store.getActions()).toMatchSnapshot();
  });
});
