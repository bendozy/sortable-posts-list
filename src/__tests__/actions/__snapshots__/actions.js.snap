// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action creators tests dispatches only all actions in the actions reducer if index 0 is selected 1`] = `
Array [
  Object {
    "lastAction": Object {
      "index": 4,
    },
    "type": "REMOVE_LAST_ACTION",
  },
  Object {
    "lastAction": Object {
      "index": 3,
    },
    "type": "REMOVE_LAST_ACTION",
  },
  Object {
    "lastAction": Object {
      "index": 2,
    },
    "type": "REMOVE_LAST_ACTION",
  },
  Object {
    "lastAction": Object {
      "index": 1,
    },
    "type": "REMOVE_LAST_ACTION",
  },
  Object {
    "lastAction": Object {
      "index": 0,
    },
    "type": "REMOVE_LAST_ACTION",
  },
]
`;

exports[`Action creators tests dispatches only one action if there is only an action Item in the actions reducer 1`] = `
Array [
  Object {
    "lastAction": Object {
      "from": 1,
      "title": "hello",
    },
    "type": "REMOVE_LAST_ACTION",
  },
]
`;

exports[`Action creators tests should create an action to dispatch last item in the actions reducer 1`] = `
Object {
  "lastAction": Object {
    "from": 1,
    "title": "hello",
  },
  "type": "REMOVE_LAST_ACTION",
}
`;
